okio/src/main/java/okio/GzipSource.kt

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:JvmName("-GzipSourceExtensions")
```

Comment:
To hide this type from Java ☹️

Reply:
And that only applies to the extension method at the bottom? Hmmm

Reply:
I assume there’s nothing special about `-`, just a sneaky way to hide it? Would `$` be even better?

Reply:
`$` is a valid character in source whereas `-` is only valid in classfiles (not source code). As a result, it's impossible to reference this type from Java and most IDEs won't bother showing it in autocomplete.